kiwi/volume_manager/btrfs.py

```diff
@@ -200,15 +205,6 @@ def get_fstab(self, persistency_type='by-label', filesystem_name=None):
         block_operation = BlockID(self.device)
         blkid_type = 'LABEL' if persistency_type == 'by-label' else 'UUID'
         device_id = block_operation.get_blkid(blkid_type)
-        if self.custom_args['root_is_snapshot']:
-            mount_entry_options = mount_options + ['subvol=@/.snapshots']
-            fstab_entry = ' '.join(
-                [
-                    blkid_type + '=' + device_id, '/.snapshots',
-                    'btrfs', ','.join(mount_entry_options), '0 0'
-                ]
-            )
-            fstab_entries.append(fstab_entry)
```

Comment:
why is it needed to remove the inclusion of the snapshots volume to the fstab file ?

Reply:
See this portion of the code:

https://github.com/OSInside/kiwi/blob/a66849ff8ea9a1c637c6dde088eebba2a8809864/kiwi/volume_manager/btrfs.py#L203-L222

For btrfs subvolumes we set fstab with the detaulf volume as `/` and the by looping to all entries in`self.subvol_mount_list`, which now also includes the `/.snapshots` subvolume if the  `btrfs_root_is_snapshot` was set. If you do not remove this portion you will end up with an fstab with two duplicated entries at the very beginning.

I think this is cleaner in the way that now `/.snapshots` is not an exception, it is just an additional subvolume automatically added when `btrfs_root_is_snapshot`.

Reply:
Ah yes makes sense it's already in the mount_list now, all clear great